modules/issue_tracker/php/endpoints/attachment.class.inc

```diff
@@ -0,0 +1,248 @@
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values = $request->getQueryParams();
+
+        $file_hash = $values['file_hash'];
+        $filename  = $values['filename'];
+        $mime_type = $values['mime_type'];
+
+        $config = \NDB_Config::singleton();
+        $attachment_data_dir = rtrim(
+            $config->getSetting('IssueTrackerDataPath'),
+            '/'
+        );
+
+        $fileToDownload = trim($attachment_data_dir) .
+            '/attachments/' .
+            $file_hash;
+
+        if (!is_readable($fileToDownload)) {
+            throw new \LorisException('Forbidden Access');
+        }
+
+        $name = rawurldecode($filename);
+
+        return (new \LORIS\Http\Response())
+            ->withHeader('Content-Type', $mime_type)
+            ->withHeader(
+                'Content-Disposition',
+                'attachment; filename=' . basename($name)
+            )
+            ->withStatus(200)
+            ->withBody(new \LORIS\Http\FileStream($fileToDownload));
+    }
+
+    /**
+     * Handle POST requests for attachment
+     * and used for uploading attachment file.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse POST request body.
+        $values     = $request->getQueryParams();
+        $user       = \NDB_Factory::singleton()->user();
+        $attachment = new \LORIS\issue_tracker\UploadHelper();
+        $response   = $attachment->setupUploading(
+            $user,
+            $_FILES,
+            $values
+        );
+        return new \LORIS\Http\Response\JsonResponse(
+            $response
+        );
+    }
+
+    /**
+     * Handle DELETE requests for attachment
+     * and marks attachment as deleted from issue_tracker.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleDELETE(ServerRequestInterface $request)
+    : ResponseInterface
+    {
+        // Verify User can delete file.
+        if ($this->_deletePermission($request)) {
+            // Parse request body.
+            $values    = $request->getQueryParams();
+            $file_hash = $values['file_hash'];
+            $factory   = \NDB_Factory::singleton();
+            $DB        = $factory->database();
+            $query     = 'UPDATE
+                             issues_attachments
+                          SET
+                             deleted = 1
+                          WHERE
+                             file_hash = :file_hash;';
+            $stmt      = $DB->prepare($query);
```

Comment:
I think you can just use `$DB->update(...)` here

Reply:
@johnsaigle I'm not sure if that should be done because from looking at the $DB->update it seems that an injection could be made from the hash.

Reply:
You're using prepared statements so it will be safe.

Reply:
@johnsaigle I'm looking at the update function in the Database.class.inc and I don't believe it uses the prepared statement. `update(string $table, array $set, array $i_where)`

Reply:
Did you look at _realupdate?

Reply:
Every database function in LORIS supports prepared statements.

Reply:
@johnsaigle didn't take a look until you mentioned it. I'm looking and see there is a prepare done. So I guess it is safe. Sorry was confused for a second.